src/diffusers/models/embeddings.py

```diff
@@ -878,3 +878,28 @@ def forward(self, x: torch.Tensor) -> torch.Tensor:
 
         latents = self.proj_out(latents)
         return self.norm_out(latents)
+
+
+class MultiIPAdapterImageProjection(nn.Module):
+    def __init__(self, IPAdapterImageProjectionLayers: Union[List[nn.Module], Tuple[nn.Module]]):
+        super().__init__()
+        self.ImageProjectionLayers = nn.ModuleList(IPAdapterImageProjectionLayers)
+
+    def forward(self, image_embeds: List[torch.FloatTensor]):
+        projected_image_embeds = []
+
+        for image_embed, ImageProjectionLayer in zip(image_embeds, self.ImageProjectionLayers):
+            if image_embed.ndim != 4:
+                raise ValueError(f"image_embeds must be 4-dimensional, got {image_embed.shape}")
+
+            batch_size, num_images, sequence_length, _ = image_embed.shape
+            image_embed = image_embed.reshape(batch_size * num_images, sequence_length, -1)
+
+            image_embed = ImageProjectionLayer(image_embed)
+
+            _, sequence_length, cross_attention_dim = image_embed.shape
+            image_embed = image_embed.reshape(batch_size, num_images, sequence_length, cross_attention_dim)
+
+            projected_image_embeds.append(image_embed)
```

Comment:
@sayakpaul about your question https://github.com/huggingface/diffusers/pull/6573/files#r1458801544

it is here - I made it so the image projection layer will return a list of `n` image embeddings, where `n` is the number of ip-adapter we loaded in